### PR TITLE
 Allow the adding of headers rather than just overwriting all headers

### LINKED ISSUE #1066: Specify different UA
In our envoirement we have by default blocked any traffic having the UA "python-requests" since we've had a bunch of DDoS attacks coming with it.

We'd love to have the possibility to set an alternate UA. 

Cheers G4bbix

### PR DESCRIPTION
**Pull request by adehad**
Replaces #958 
- [x] add a simple test
- [x] add to examples, fixes #1066

### CODE REVIEW

**Comment by adehad** [comment]
@studioj how does this look?

File: tests/test_client.py [hunk 1]
```diff
@@ -119,3 +129,53 @@ def test_result_list_if_empty():
...
+@pytest.mark.parametrize(
+    "options_arg",
+    [
+        {"headers": {"Content-Type": "application/json;charset=UTF-8"}},
+        {"headers": {"random-header": "nice random"}},
+        {},
+    ],
+    ids=["overwrite", "new", "none"],
+)
+def test_headers_unclobbered_update(options_arg, no_fields):
```
Comment (studioj) [inline tests/test_client.py:143]: its a matter of preference, i dont like parameterized tests too much certainly if you change your test code to differ per parameter.

its opening up for weirdness during refactoring.
does it hurt to split up in 2 or more tests? so the test code doesnt need to change regarding which parameter is passed?

**Review by studioj — COMMENTED** [review]
for the rest it seems ok... 
although the reason why to add this is kinda unclear from the code you;re adding
maybe add an example in hte examples?

File: tests/test_client.py [hunk 2]
```diff
@@ -55,6 +55,16 @@ def remove_by_slug():
...
 
 
+@pytest.fixture()
+def no_fields(monkeypatch):
+    """When we want to test the __init__ method of the jira.client.JIRA
+    we don't need any external calls to get the fields.
+
+    We don't need the features of a MagicMock, hence we don't use it here.
+    """
+    monkeypatch.setattr(jira.client.JIRA, "fields", lambda *args, **kwargs: [])
```
[the remark this answers is not in the document]
Comment (adehad) [reply tests/test_client.py:65]: let me know if I can help explain any queries you have around fixtures. 

The simplest way I can describe them is they are a function called before your test (e.g. similar to `setUp()`), and one way you tell a test to use them is to use the name of your fixture (basically a function) as a keyword argument in your test.

This particular fixture is using another (inbuilt) fixture called `monkeypatch` that works somewhat like the `mock.patch()` style patching.

**Review by studioj — APPROVED** [review]

**Comment by studioj** [comment]
@ssbarnea ready for merge :-)

**Pull request merged**